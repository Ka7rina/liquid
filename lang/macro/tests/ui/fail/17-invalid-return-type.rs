use liquid_lang as liquid;

#[derive(Clone, Copy)]
pub struct I(i32);

#[liquid::contract(version = "0.1.0")]
mod noop {
    use super::*;

    #[liquid(storage)]
    struct Noop {}

    #[liquid(methods)]
    impl Noop {
        pub fn constructor(&mut self) {}

        pub fn noop(&self) -> I {}
    }
}

fn main() {}
