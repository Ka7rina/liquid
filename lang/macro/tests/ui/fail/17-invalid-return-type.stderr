error[E0277]: the trait bound `I: liquid_lang::The_Type_You_Used_Here_Must_Be_An_Valid_Liquid_Data_Type` is not satisfied
  --> $DIR/17-invalid-return-type.rs:18:24
   |
18 |         fn noop(&self) -> I {}
   |                        ^^^^ the trait `liquid_lang::The_Type_You_Used_Here_Must_Be_An_Valid_Liquid_Data_Type` is not implemented for `I`
